### PR TITLE
Fix field naming in index_option

### PR DESCRIPTION
**Pull request by zonca**
- Move set_fields from RunRead to AttrParse2 in AwkwardCompiler.scala

### CODE REVIEW

**Comment by zonca** [comment]
In the test case of `index_option`:
https://github.com/ManasviGoyal/kaitai_struct_awkward_runtime/blob/5e71bafddc7c68f3a4bf103a8fe2bf6fa08cd3b0/tests/test_index_option.py

we see that `expected_array_bank1` has fields named "0", "1", "2"

@ManasviGoyal do you confirm we want them named with their actual names? what about instead having an empty array, given `bank1` is not in the file?

File: shared/src/main/scala/io/kaitai/struct/languages/AwkwardCompiler.scala [hunk 1]
```diff
@@ -727,6 +726,7 @@ class AwkwardCompiler(
           s"${if (unionIndex.contains("child")) ".content<" + unionIndex.split("_").last + ">()" else ""}" + 
           s";"
         )
+        outSrc.puts(s"${nameList.last}_builder.set_fields(${nameList.last}_fields_map);")
```
Comment (zonca) [inline shared/src/main/scala/io/kaitai/struct/languages/AwkwardCompiler.scala:729]: I tried moving `set_fields` into `AttrParse2`, but this actually breaks the test on the header.
While it used to parse the header as:

    array([(100, 16)], dtype=[('bank_hdrA__Zall_bank_size', '<u4'), ('bank_hdrA__Zflags', '<u4')])

now it lost the field names:

    array([(100, 16)], dtype=[('0', '<u4'), ('1', '<u4')])

here is the generated cpp
https://gist.github.com/zonca/425cc2ab0489949466b07eb1582013b2

Comment (ManasviGoyal) [reply shared/src/main/scala/io/kaitai/struct/languages/AwkwardCompiler.scala:729]: @zonca  I have added a fix in #6. Please check and let me know if it works.

Comment (ManasviGoyal) [reply shared/src/main/scala/io/kaitai/struct/languages/AwkwardCompiler.scala:729]: In Awkward Arrays, we have `bank1` as `None`
```
>>> awkward_array.to_list()
[{'index_optionA__Zbank_header': {'bank_hdrA__Zall_bank_size': 100, 'bank_hdrA__Zflags': 16}, 'index_optionA__Zbank1': None, 'index_optionA__Zbank2': {'bank32A__Zname': 'Bank', 'bank32A__Ztype': 655361, 'bank32A__Zsize': 1802395970}}]
```
Now we can see the fields of `bank1
```
>>> awkward_array.layout.form
RecordForm([RecordForm([NumpyForm('uint32'), NumpyForm('uint32')], ['bank_hdrA__Zall_bank_size', 'bank_hdrA__Zflags']), IndexedOptionForm('i64', RecordForm([ListOffsetForm('i64', NumpyForm('uint8', parameters={'__array__': 'char'}), parameters={'__array__': 'string'}), NumpyForm('uint16'), NumpyForm('uint16')], ['bank16A__Zname', 'bank16A__Ztype', 'bank16A__Zsize'])), IndexedOptionForm('i64', RecordForm([ListOffsetForm('i64', NumpyForm('uint8', parameters={'__array__': 'char'}), parameters={'__array__': 'string'}), NumpyForm('uint32'), NumpyForm('uint32')], ['bank32A__Zname', 'bank32A__Ztype', 'bank32A__Zsize']))], ['index_optionA__Zbank_header', 'index_optionA__Zbank1', 'index_optionA__Zbank2'])
```

**Comment by zonca** [comment]
see #6

**Pull request closed without merging**